pkg/handlers/archive.go

```diff
@@ -240,6 +245,6 @@ func (h *archiveHandler) extractorHandler(dataOrErrChan chan DataOrErr) func(con
 		h.metrics.observeFileSize(fileSize)
 
 		lCtx.Logger().V(4).Info("Processed file successfully", "filename", file.Name(), "size", file.Size())
-		return h.openArchive(lCtx, depth, rdr, dataOrErrChan)
+		return h.openArchive(lCtx, depth, (archiveEntryPath + "/" + file.Name()), rdr, dataOrErrChan)
```

Comment:
Do we care about OS-specific paths here? If so, using [filepath.Join](https://pkg.go.dev/path/filepath#Join) would be a better option.

Reply:
Definitely a better choice. Will update.